### PR TITLE
change: v4 Xcode 15 fixes

### PR DESCRIPTION
**Pull request by goergisn**
## Changes

<change>

* Added support for [Xcode 15](https://developer.apple.com/xcode/) 
  * `@Observable` renamed to `@AdyenObservable`.
  * iOS 11 is no longer supported because it's [not supported by Xcode 15](https://developer.apple.com/support/xcode/).

</change>

### CODE REVIEW

**Comment by chuck-gnaw** [comment]
hey Adyen team, could you please merge the changes to `V4` as soon as possible? our team will upgrade Xcode to 15 soon.

**Comment by erenbesel** [comment]
> hey Adyen team, could you please merge the changes to `V4` as soon as possible? our team will upgrade Xcode to 15 soon.

@chuck-gnaw we are releasing a non-breaking v4 (4.10.5) version this week, after which we will make the forced XCode changes and release 4.11.0, hopefully coming weeks

**Review by erenbesel — DISMISSED** [review]

**Review by neelSharma12 — DISMISSED** [review]

File: Demo/Configuration.swift [hunk 1]
```diff
@@ -8,7 +8,7 @@ import Adyen
 import Foundation
 import PassKit
 
-internal enum ConfigurationConstants {
+enum ConfigurationConstants {
```
Comment (descorp) [inline Demo/Configuration.swift]: This is due to a Swiftformat update..
Pull latest `develop` and latest `Internal`
Also you need to revert those changes in `Demo/Configuration.swift`

File: MIGRATION.md [hunk 2]
```diff
@@ -34,3 +34,6 @@ func didFail(with error: Error, from component: PaymentComponent) {
 - Make `countryCode` non-optional for `Payment`.
 - Add assert check on `Payment` init to validate country code.
 - Move individual `ActionComponent` styles into one struct `ActionComponentStyle`, which is an instance variable in `DropInComponent.Style`.
+
+## 4.10.6
+- `Observable` has been renamed to `AdyenObservable` to avoid name collision with `@Observable` swift macro.
```
Comment (descorp) [inline MIGRATION.md:39]: 👍

**Review by erenbesel — APPROVED** [review]

**Review by descorp — APPROVED** [review]

**Comment by piercifani** [comment]
Any ETA on this? this is blocking our iOS 17 release!

**Comment by neelSharma12** [comment]
@piercifani We will release v4.11.0 by next week.

**Pull request merged**